lib/dfe/analytics/event.rb

```diff
@@ -73,7 +73,8 @@ def with_entity_table_name(table_name)
       end
 
       def with_data(hash)
-        @event_hash.deep_merge!(data: hash_to_kv_pairs(hash))
+        @event_hash.deep_merge!(data: hash_to_kv_pairs(hash[:data]))
+        @event_hash.deep_merge!(hidden_data: hash_to_kv_pairs(hash[:hidden_data]))
 
```

Comment:
I guess the only downside to doing it this way is we now have a more complex data structure that might be a bit less performant